design/specs/table-import.md

```diff
@@ -14,153 +14,175 @@ Adding a new table from a file import allows users to input data into Mathesar w
-- Copy will be revised later with the purpose of supporting best practices while guiding the user during the table creation process. We want to be more explicit in regards to goals for every step. 
+User can paste a URI in this block. If the URI is valid, the user would be able to click on the continue button to move to the processing import view.
 
-#### Implementation Note:
-- Empty table flow will be designed in a future iteration.
+##### Paste
 
-### User Selects Data Import Method
-Once the new table is created, The system will prompt the user to start the import data process. There are various options to import data, whether from a local file, clipboard, or remote location.
-By default, Upload File is the selected option.
+User can paste data directly here in this block. If the data pasted here is valid, the user would be able to click on the continue button to move to the processing import view.
 
-![](/assets/design/specs/table-import/zU9Csfe.png)
 
+#### Import Options
 
-### User Uploads a File
-The user then proceeds to upload a CSV file, and once it's completed, the system will render the file along with an option to remove it and upload another file.
+There are a few flags that user will find helpful for the import process. And they are present on the upload view itself. 
 
-![](/assets/design/specs/table-import/1FZ5Efd.png)
+##### Data Type Inference
 
-In most cases, when the file is large, processing will take longer. For that reason, a loading UI state is needed.
+This is a checkbox input. So when the checkbox is checked, the data type inference will not happen while the import process. This would make the import process faster and in some cases, the process possible too. The user would then have to select the data types in the confirmation stage. 
 
-![](/assets/design/specs/table-import/By4JP0CK_.png)
+If not checked, the data type will be detected and this might take a longer time and sometimes the import might not even happen. The user can still change the data types in the confirmation stage.
 
-If any errors prevent the file from being uploaded, an error message will be displayed.
+##### Skip Warnings
 
-![](/assets/design/specs/table-import/HJxJY0AY_.png)
+This is again a checkbox, if checked he user would not be notified with any warnings related to the import process. This way when there are situations, the defaults will be considered and the user won't be informed about it.
 
-#### Implementation Notes
-- Paste and From URL methods will be designed in a future iteration. The current use case only uses the file import method. 
+##### Automatic Delimiter Overwrite
 
-### User Confirms Data
-If the system successfully processed the file, it will prompt the user to confirm if a preview of their data looks correct. The user can also choose to ignore columns, rename them, and change data types from this view. 
+This is an input field where user can enter a deliminator used in their file. The default is taken as a comma(,). 
```

Comment:
As per a discussion with @ghislaineguerin, we discussed that sometimes there might be JSON data which might confuse the algorithm and hence the file would not be imported. So that is why I thought this might be a better way to handle it.

Reply:
This is probably going to confuse our user persona (same as my comments above). It seems like showing an error for un-importable files would work fine.

Reply:
Sorry, I'm pretty late to this convo; just catching up. It's possible to choose the incorrect delimiter and still have a valid import set up.

Imagine (if you will) a headerless TSV whose correct delimiter is "tab". Now, further suppose there is a JSON column in this TSV, with a fixed schema. In particular, it has the same number of commas on each row. Our algo will not be able to choose properly which delimiter to use, and in fact, will chose comma in that case. We could change that specific behavior (i.e., put tab above comma in the choice hierarchy), but I don't think that solves the problem in general.

I think we _should_ display and give the option to modify the delimiter which has been chosen by the algorithm, but we shouldn't assume it's a comma unless that's what's returned by the algo. I think people who use spreadsheets regularly should be okay with that, since it's a common step for importing data into spreadsheet software.

@kgodey Are you arguing that they shouldn't have a chance to change the delimiter, or am I misunderstanding?

Reply:
Okay, now I've taken a moment to read through the document and see this in context. I agree with @kgodey that the user shouldn't have to make this choice at the upload/initial stage. I think it's more appropriate to give the chance to modify at the confirmation stage if something has gone wrong.